File: strategy/marvin/agent/hidden/default.lua
local Base = require "agent/base/behavior"
local Default = Class("Agent.Hidden.Default", Base)

local World = require "../base/world"
local RescueRobot = require "task/rescuerobot"


function Default:check()
	return true
end

function Default:_updateTask()
	return RescueRobot
end

return Default

File: strategy/typescript/glados/agent/hidden/default.ts
import { Behavior, TaskAssignment } from "glados/agent/base/behavior";
import { RescueRobot } from "glados/task/hidden/rescuerobot";


export class Default extends Behavior {
	public check(): Behavior {
		return this;
	}

	protected _updateTask(): TaskAssignment<typeof RescueRobot> {
		return [RescueRobot];
	}
}
